File: src/types/Coords.go
package types

type Coords struct {
	X int `yaml:"x,omitempty"`
	Y int `yaml:"y,omitempty"`
}

type CoordsArea struct {
	X int `yaml:"x,omitempty"`
	Y int `yaml:"y,omitempty"`
	W int `yaml:"w,omitempty"`
	H int `yaml:"h,omitempty"`
}

File: rust/src/models/Coords.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct Coords {
    pub x: i32,
    pub y: i32,
}

impl Coords {
    fn new(x: i32, y: i32) -> Coords {
        Coords { x, y }
    }
}

#[derive(Serialize, Deserialize, Debug)]
pub struct CoordsArea {
    pub x: i32,
    pub y: i32,
    pub w: i32,
    pub h: i32,
}

impl CoordsArea {
    fn new(x: i32, y: i32, w: i32, h: i32) -> CoordsArea {
        CoordsArea { x, y, w, h }
    }
}